status_handler_test.go

```diff
@@ -129,5 +133,91 @@ func TestValidationStatusHandler(t *testing.T) {
 	assert.NoError(t, res.Body.Close())
 	require.NoError(t, err)
 
-	assert.Equal(t, "{\"error\":{\"body\":{\"fields\":{\"field\":{\"errors\":[\"The field is required\"]}},\"errors\":[\"The body is required\"]},\"query\":{\"fields\":{\"query\":{\"errors\":[\"The query is required\"]}}}}}\n", string(body))
+	assert.Equal(t, `{"error":{"body":{"fields":{"field":{"errors":["The field is required"]}},"errors":["The body is required"]},"query":{"fields":{"query":{"errors":["The query is required"]}}}}}`+"\n", string(body))
+}
+
+func TestParseErrorStatusHandler(t *testing.T) {
+	tests := []struct {
+		name            string
+		err             error
+		expectedMessage string
+		expectedStatus  int
+	}{
+		{
+			name:            "InvalidJSONBody",
+			err:             ErrInvalidJSONBody,
+			expectedMessage: "The request Content-Type indicates JSON, but the request body is empty or invalid.",
+			expectedStatus:  http.StatusBadRequest,
+		},
+		{
+			name:            "InvalidQuery",
+			err:             ErrInvalidQuery,
+			expectedMessage: "Failed to parse query string due to invalid syntax or unexpected input format.",
+			expectedStatus:  http.StatusBadRequest,
+		},
+		{
+			name:            "InvalidContentForType",
+			err:             ErrInvalidContentForType,
+			expectedMessage: "The request content does not match its type. E.g. invalid multipart/form-data or a problem with the file upload.",
+			expectedStatus:  http.StatusBadRequest,
+		},
+		{
+			name:            "ErrorInRequestBody",
+			err:             ErrErrorInRequestBody,
+			expectedMessage: "Failed to read request body due to connection issues, timeouts, size mismatches, or corrupted data.",
+			expectedStatus:  http.StatusBadRequest,
+		},
+		{
+			name:            "OtherError",
+			err:             errors.New("some.other.error"),
+			expectedMessage: "some.other.error",
+			expectedStatus:  http.StatusBadRequest,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			req, resp, recorder := prepareStatusHandlerTest()
+
+			handler := &ParseErrorStatusHandler{}
+			handler.Init(resp.server)
+
+			req.Extra[ExtraParseError{}] = tt.err
+			resp.Status(tt.expectedStatus)
+
+			handler.Handle(resp, req)
+
+			res := recorder.Result()
+			body, err := io.ReadAll(res.Body)
+
+			assert.NoError(t, res.Body.Close())
+			require.NoError(t, err)
+
+			expectedResponse := fmt.Sprintf(`{"error":"%s"}`, tt.expectedMessage) + "\n"
+			assert.Equal(t, expectedResponse, string(body))
+			assert.Equal(t, tt.expectedStatus, res.StatusCode)
+		})
+	}
+}
+
+func TestParseErrorStatusHandlerWithNonSupportedErrorStruct(t *testing.T) {
+	req, resp, recorder := prepareStatusHandlerTest()
+
+	handler := &ParseErrorStatusHandler{}
+	handler.Init(resp.server)
+
+	req.Extra[ExtraValidationError{}] = errors.New("some.other.error")
```

Comment:
Remove this line and rename the test with `TestParseErrorStatusHandlerWithoutExtra`.